tools/docker/builder/openwrt/Dockerfile

```diff
@@ -0,0 +1,100 @@
+ARG HOME=/home/openwrt
+
+####
+## OpenWrt pre-requisites and "openwrt" user
+####
+FROM ubuntu:18.04 as openwrt-prerequisites
+
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends \
+    build-essential libncurses5 libncurses5-dev python python3 \
+    unzip git ca-certificates gawk wget file bash \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN useradd -ms /bin/bash openwrt
+
+####
+## Pre-built OpenWrt (and extracted sdk)
+####
+FROM openwrt-prerequisites as openwrt-prebuilt
+
+ARG HOME
+COPY --chown=openwrt:openwrt configs/sdk_config $HOME/sdk_config
+USER openwrt
+
+ARG OPENWRT_REPOSITORY=https://git.prpl.dev/prplwrt/prplwrt.git
+ARG OPENWRT_VERSION=9d2efd
+ARG TARGET=mvebu
+
+WORKDIR $HOME
+
+RUN git clone "$OPENWRT_REPOSITORY" openwrt \
+    && cd openwrt \
+    && git checkout "$OPENWRT_VERSION"
+
+WORKDIR $HOME/openwrt
+
+RUN cp "${HOME}/sdk_config" .config \
+    && echo "CONFIG_TARGET_${TARGET}=y" >> .config \
+    && cat .config \
+    && make defconfig \
+    && make -j$(nproc) 2>&1 | tee build.log
+
+RUN TAR="$(find bin/targets -name 'openwrt-sdk*' -exec echo {} \;)" \
+    && mkdir -p "${HOME}/openwrt_sdk" \
+    && mv -v "$TAR" "${HOME}/openwrt_sdk" \
+    && TAR="$(basename $TAR)" \
+    && cd "${HOME}/openwrt_sdk" \
+    && tar -xf "$TAR" --strip-components=1 \
+    && rm "$TAR"
+
+####
+## Prebuilt sdk only
+####
+FROM openwrt-prerequisites as openwrt-sdk
+ARG HOME
+COPY --from=openwrt-prebuilt --chown=openwrt:openwrt $HOME/openwrt_sdk $HOME/openwrt_sdk
+WORKDIR $HOME/openwrt_sdk
+
+####
+## Prebuilt sdk and prplmesh dependencies (and some debug tools)
+####
+
+FROM openwrt-sdk as prplmesh-builder
+
+ARG HOME
+COPY --chown=openwrt:openwrt configs/prplmesh_config $HOME/prplmesh_config
+ARG PRPLMESH_VERSION=82f1e83
+ARG TARGET_PLATFORM_TYPE=turris-omnia
+ARG PRPL_FEED=https://git.prpl.dev/prplmesh/feed-prpl.git
+# Depending on the target, you may want to use dwpal instead of nl80211:
+ARG USE_DWPAL=false
+
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends \
+    python3 python-yaml rsync less vim libgetopt-complete-perl \
+    && rm -rf /var/lib/apt/lists/*
+    # Building tlvfs currently uses the docker
+    # container's python (not the one from OpenWrt)
+
+USER openwrt
+
+RUN scripts/feeds update -a \
+    && scripts/feeds install -a \
+    && echo "src-git prpl ${PRPL_FEED}" >> feeds.conf.default \
```

Comment:
If you do this at the beginning, you don't need the additional update below.

Reply:
Done.